docs/java/features/odata/type-safe-client-odata-v4.mdx

```diff
@@ -211,45 +211,37 @@ The described API is subject to change
 #### Execute Batch Request
 The `service` object offers the method `batch` which allows access to the batch-related methods that help build the batch request.
 
-Multiple single requests (like `createBusinessPartnerAddress`) can be wrapped into so-called change sets.
-Each change set is opened with `beginChangeSet` and closed with `endChangeSet`. You can wrap multiple change sets into one batch request.
+Multiple single requests (like `createBusinessPartnerRequest`) can be wrapped into so-called change sets.
+A change set can be added by `addChangeset` and the corresponding create, update or delete operations can be passed as partmenters into it. You can wrap multiple change sets into one batch request.
+Similarly, retrieve operations can be added to the batch request by using `addReadOperations`.
 
-Use `executeRequest` to issue the batch request to the remote system. We receive an instance of `BatchResponse` as result object.
+Use `execute` to issue the batch request to the remote system. We receive an instance of `BatchResponse` as result object.
 
 ```java
 final BatchResponse result =
```

Comment:
How about prepending the expected types...?

```suggestion
GetAllRequestBuilder<BusinessPartner> getBusinessPartners = service.getAllBusinessPartner();
GetByKeyRequestBuilder<BusinessPartner> getBusinessPartnerByKey = ...
CreateRequestBuilder<BusinessPartner> createBusinessPartnerRequest1 = ...
CreateRequestBuilder<BusinessPartner> createBusinessPartnerRequest2 = ...
DeleteRequestBuilder<BusinessPartner> deleteBusinessPartnerRequest = ...

final BatchResponse result =
```

Reply:
yes it makes it more clear.